### PR TITLE
Fixes PR

### LINKED ISSUE #5507: Not Really an Issue, Just a Suggestion. [Regarding the Piano Scene]
Okay so, I didn't know Monika was going to play her song once I opened up the game again. I had my music turned all the way down in the game, so I was pretty much left with awkwardly staring with a piano covering half her face. Next time, could you guys either let her turn the music option all the way up for the player or at least let her ask if it's turned on? Thank you!

### LINKED ISSUE #5513: The piano doesn´t look right when zoomed in.
The Piano, that gets used when you ask Monika to play "Your Reality" doesn´t seem to work visually when you are zoomed in. I noticed that because I always have Monika all the way up close with the zoom option in the Extras-Menu, and that led to her head being almost as big as the piano itself when she played "Your Reality"!

My suggestion on how to fix this is to have it automatically zoom out to the needed extent, like it does when she gets up to get coffee, before she gets the piano on screen. 

Thank you for your time~!

### PR DESCRIPTION
**Pull request by jmwall24**
Cleaning up the goofs we made in 0.11.0.
Closes #5513, closes #5507

## Fixes

- piano scenes looking bad due to Monika being zoomed. we now reset zoom to default in these scenarios

- not being able to hear Monika play/sing during the piano scenes and having no way to turn up the in-game volume once the piano part starts. we now warn the player what is coming, remind them to turn up the volume if it's set to zero, give them the option not listen right now in case they can't, and also made it so the volume adjust hotkeys are never shielded.

- the twintails hair style being locked for people that didn't get the Miku costume on Halloween, even if they've installed the sprite pack.

- a reoccurring issue where the gender change and/or player name change topics can become locked for various reasons.

- `monika_being_virtual` being added to the calendar when the start_date is set.

## Additions and changes

- adds `screen hot_keys()` a new screen in the escape menu that lists the most useful hotkeys

- `mas_HKRaiseShield_main(`) and `mas_HKDropShield_main()` now control the Talk, Extra, Music and Play hotkeys, but not the music controller hotkeys.

- `mas_HKRaiseShield()` and `mas_HKDropShield()` control everything aforementioned `main()` functions do, plus the music controller hotkeys.

- creates `mas_RaiseShield_piano()` and `mas_DropShield_piano()` added specifically for piano scenes.

- creates `mas_MUINRaiseShield()` and `mas_MUINDropShield()` which control all music-based interactions.

- creates `hasMusicMuted()` that checks whether the music channel is muted or the 'Mute All' option is enabled

## Testing

- verify `monika_credits_song` asks you if you have time to listen to the songs right now, and if yes, verify she reminds you to turn your speakers and volume up, and if music volume is still set to 0 or the mute all option is enabled, verify she informs you of this. 

- verify when the songs play (via credits_song or the pool topics) that zoom is reset and the volume adjust hotkeys work throughout the songs.

- verify all shields are raised and lowered as expected via all ways to play the songs.

- verify the `Hotkeys` screen is present in the escape menu and that it looks ok.

-  push the `greeting_tears` event and verify that `monika_being_virtual` now has a start_date and that it **is not** on the calendar.

- on a persistent that did not see the Miku costume on Halloween, install the twintails spritepack, `call v0_11_1` and verify the twintails hairstyle appears in the hair selector.

### CODE REVIEW

**Comment by ThePotatoGuy** [comment]
>a reoccurring issue where the gender change and/or player name change topics can become locked for various reasons.

how was this issue found and did we find a root cause?

**Comment by multimokia** [comment]
> how was this issue found and did we find a root cause?

Issues are from translated versions, additionally event data wipes from the 0.8.14 -> 0.9.0/0.9.1 releases.

File: Monika After Story/game/screens.rpy [hunk 1]
```diff
@@ -1484,6 +1486,67 @@ screen notif_settings():
...
+                    text _("S")
+                    text _("Esc")
+
+            hbox:
+                style_prefix mas_ui.cbx_style_prefix
+                vbox:
+                    label _("Music")
+                    spacing 10
+                    text _("Vol Up")
+                    text _("Vol Down")
```
Comment (ThePotatoGuy) [inline Monika After Story/game/screens.rpy:1533]: can you spell out "Volume" here for clarity? We have enough space to use the full word anyway.

Comment (jmwall24) [reply Monika After Story/game/screens.rpy:1533]: Ah yeah, I did that originally when I made the menu a different way, but should fit now.

File: Monika After Story/game/script-story-events.rpy [hunk 2]
```diff
@@ -2268,51 +2268,65 @@ label monika_credits_song:
...
-        else:
-            m 3eua "Hey [player], I finally finished that song I've been working on for you."
+                m 1ekbsa "Ehehe~"
+                show monika 5ekbsa at t11 zorder MAS_MONIKA_Z with dissolve
+                m 5ekbsa "Thank you for coming back to me my love."
+
+            "Sorry, I can't right now.":
+                m 3ekd "Oh, okay."
+                m 1eka "That's fine [player], I understand if you don't have the time or just can't listen to music right now."
+                m 3hua "Just me know when it's a better time for you and I'll happily play it for you then~"
```
Comment (MisterSimple) [inline Monika After Story/game/script-story-events.rpy:2294]:
```suggestion
                m 3hua "Just let me know when it's a better time for you and I'll happily play it for you then~"
```

**Pull request merged**